packages/block-editor/src/components/block-list/block-list-context.native.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+export const DEFAULT_BLOCK_LIST_CONTEXT = {
+	scrollRef: null,
+	blocksLayouts: { current: {} },
+	updateBlocksLayouts,
+};
+
+const Context = createContext( DEFAULT_BLOCK_LIST_CONTEXT );
+const { Provider, Consumer } = Context;
+
+/**
+ * Finds a block's layout data within the provided data object.
```

Comment:
Nitpick: I'd consider mentioning that the find is made by the block's client Id.
```suggestion
 * Finds a block's layout data by its client Id.
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/39089/commits/b1ee8f60d54f96f3c5e723b1de903928357122da